DAC/src/ringbuffer.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * ringbuffer.c
+ *
+ *  Created on: Feb 6, 2018
+ *      Author: sharana
+ *      Description: Program to initiate a ring buffer.
+ */
+#include"ringbuffer.h"
+#include<inttypes.h>
+/*
+ * Here we have a method put characters into the buffer.
+ * @param buffer it is a structure type to access the correct ring buffer.
+ * @param element it is the character.
+ */
+void put(RingBuffer* buffer, char element){
+	//Check space
+	 if(hasSpace(&buffer)== 1)
```

Comment:
When determining something that should be something like a "bool", it is better to not check for a hard coded 1. Technically, 0 is false, but _any_ non-zero value can be interpreted as true.